From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lukasz Sobala <lsobala1@gmail.com>
Date: Tue, 18 Aug 2026 21:38:48 +0200
Subject: arm64: dts: rockchip: rock-2f: Enable PCIe

Enable PCIe Gen2 x1 for the PCIe FPC connector. The combo PHY is
already enabled in rk3528-rock-2.dtsi and is free for PCIe because the
Type-C port is USB 2.0 only. Reset GPIO and the 3V3 rail match
rk3528-rock-2f.dts in the Radxa 6.1 kernel.

Signed-off-by: Lukasz Sobala <lsobala1@gmail.com>
---
diff --git a/arch/arm64/boot/dts/rockchip/rk3528-rock-2f.dts b/arch/arm64/boot/dts/rockchip/rk3528-rock-2f.dts
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/rockchip/rk3528-rock-2f.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3528-rock-2f.dts
@@ -7,6 +7,21 @@
 / {
 	model = "Radxa ROCK 2F";
 	compatible = "radxa,rock-2f", "rockchip,rk3528";
+
+	vcc3v3_pcie20: regulator-3v3-vcc-pcie20 {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc3v3_pcie20";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		startup-delay-us = <5000>;
+		vin-supply = <&vcc5v0_sys>;
+	};
+};
+
+&pcie {
+	reset-gpios = <&gpio1 RK_PA2 GPIO_ACTIVE_HIGH>;
+	vpcie3v3-supply = <&vcc3v3_pcie20>;
+	status = "okay";
 };
 
 &usb_host0_xhci {
-- 
Armbian

